lib/ex_unit/lib/ex_unit/runner.ex

```diff
@@ -316,13 +420,17 @@ defmodule ExUnit.Runner do
   end
 
   defp exec_on_exit(test_or_case, pid, timeout) do
-    case ExUnit.OnExitHandler.run(pid, timeout) do
-      :ok ->
-        test_or_case
-
-      {kind, reason, stack} ->
-        state = test_or_case.state || failed(kind, reason, prune_stacktrace(stack))
-        %{test_or_case | state: state}
+    if ExUnit.OnExitHandler.registered?(pid) do
```

Comment:
Why do we need to wrap this on `registered?`? `run` should return `:ok` if there is nothing registered. We probably don't need a pre-check.